src/MoBi.Presentation/Tasks/SimulationCommitTask.cs

```diff
@@ -0,0 +1,126 @@
+using System.Collections.Generic;
+using System.Linq;
+using MoBi.Core.Commands;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Helper;
+using MoBi.Core.Services;
+using OSPSuite.Core.Commands.Core;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+using static MoBi.Assets.AppConstants.Commands;
+
+namespace MoBi.Presentation.Tasks
+{
+   public interface ISimulationCommitTask
+   {
+      IMoBiCommand CommitSimulationChanges(IMoBiSimulation simulationWithChanges);
+   }
+
+   public class SimulationCommitTask : ISimulationCommitTask
+   {
+      private readonly IMoBiContext _context;
+      private readonly ITemplateResolverTask _templateResolverTask;
+      private readonly IEntitiesInSimulationRetriever _entitiesInSimulationRetriever;
+      private readonly IInitialConditionsCreator _initialConditionsCreator;
+      private readonly IParameterValuesCreator _parameterValuesCreator;
+      private readonly INameCorrector _nameCorrector;
+
+      public SimulationCommitTask(IMoBiContext context,
+         ITemplateResolverTask templateResolverTask,
+         IEntitiesInSimulationRetriever entitiesInSimulationRetriever,
+         IInitialConditionsCreator initialConditionsCreator,
+         IParameterValuesCreator parameterValuesCreator,
+         INameCorrector nameCorrector)
+      {
+         _context = context;
+         _templateResolverTask = templateResolverTask;
+         _entitiesInSimulationRetriever = entitiesInSimulationRetriever;
+         _initialConditionsCreator = initialConditionsCreator;
+         _parameterValuesCreator = parameterValuesCreator;
+         _nameCorrector = nameCorrector;
+      }
+
+      public IMoBiCommand CommitSimulationChanges(IMoBiSimulation simulationWithChanges)
+      {
+         var lastModuleConfiguration = simulationWithChanges.Configuration.ModuleConfigurations.Last();
```

Comment:
Can we add a comment in the code explaining why this is last? This is a decision we made but this will be forgotteh

Reply:
I'll make a comment at the public interface level describing how commits work